shopfloor/services/zone_picking.py

```diff
@@ -535,23 +552,34 @@ def _scan_source_location(
             )
         else:
             # if no move line, narrow the list of move lines on the scanned location
-            response = self._list_move_lines(location)
-            message = self.msg_store.location_empty(location)
+            response = self._list_move_lines(location, package=package)
+            # Location is empty: invite the user to scan a package if the option is enabled
+            if scan_location_or_pack_first:
+                message = self.msg_store.location_empty_scan_package(location)
+            else:
+                message = self.msg_store.location_empty(location)
         return response, message
 
-    def _find_product_in_location(self, location, product_id):
+    def _find_product_in_location(self, location, product_id, package=False):
         """Find the prooducts in stock in given location move line in the location."""
         domain = [("location_id", "=", location.id)]
         if product_id:
             domain.append(("product_id", "=", product_id))
+        if package:
+            domain.append(("package_id", "=", package.id))
         quants = self.env["stock.quant"].search(domain)
         product = quants.product_id
         lot = quants.lot_id
         package = quants.package_id
         return product, lot, package
 
     def _scan_source_package(
-        self, barcode, confirmation=False, product_id=False, sublocation=False
+        self,
+        barcode,
+        confirmation=False,
+        product_id=False,
+        sublocation=False,
+        package=False,
```

Comment:
This method is lacking a first check like there is in _scan_source_location
After `if not package`

```
        if not package.location_id.is_sublocation_of(self.zone_location):
            response = self._list_move_lines(self.zone_location)
            message = self.msg_store.location_not_allowed()
            return response, message
```

and below in the `Check if the package selected can be a substitute on a move line`, the sublocation check can be removed as done in any case at the beginning with this new addition

Reply:
Done